doc/JSON_ROOF_MAPGEN.md

```diff
@@ -0,0 +1,179 @@
+      "place_nested": [
+        {
+          "chunks": [
+            [ "null", 50 ],
+            [ "roof_4x4_party", 15 ],
+            [ "roof_4x4_holdout", 5 ],
+            [ "roof_4x4_utility", 40 ],
+            [ "roof_5x5_coop", 5 ]
+          ],
+          "x": [ 3, 15 ],
+          "y": [ 3, 7 ]
+        }
+      ]
+    }
+}
+````
+
+## Linking the main floor and the roof
+Navigate to `json/overmap/multitile_city_buildings.json` or `json/overmap/multitile_buildings_terrain.json` for buildings taking up more then one overmap tile per z level (schools, mansions).
+Add an entry for the main floor.  The `point` coordinates define to the x, y, z positions of the building.  The 1 places the roof one z level above the ground floor.
+Append north for rotating buildings to orient the z levels.
+
+````json
+  {
+    "type": "city_building",
+    "id": "abstorefront",
+    "locations": [ "land" ],
+    "overmaps": [
+      { "point": [ 0, 0, 0 ], "overmap": "abstorefront_north" },
+      { "point": [ 0, 0, 1 ], "overmap": "abstorefront_roof_north" }
+    ]
+  }
+````
+## Overmap Specials
+
+Overmap specials are handled a little differently.  They use `json/overmap/specials.json` for both linking z levels and overmap spawning.  A special won't need an entry in `data/json/regional_map_settings.json`
+
+Overmap special example:
+````json
+  {
+        "type" : "overmap_special",
+        "id" : "Evac Shelter",
+        "overmaps" : [
+            { "point":[0,0,0], "overmap": "shelter"},
+            { "point":[0,0,-1], "overmap": "shelter_under"},
+            { "point":[0,0,1], "overmap": "shelter_roof"}
+        ],
+        "connections" : [
+            { "point" : [0,-1,0], "terrain" : "road" }
+        ],
+        "locations" : [ "wilderness" ],
+        "city_distance" : [5, 10],
+        "city_sizes" : [4, 12],
+        "occurrences" : [1, 3],
+        "rotate" : false,
+        "flags" : [ "CLASSIC" ]
+  }
+````
+## Adding the overmap_terrain entry
+
+Navigate to `data/json/overmap_terrain.json`
+Every z level gets an entry that defines how it appears on the overmap.
+The name field will determine what is displayed in the in-game overmap.
+The entries should share the same color and symbol.
+
+````json
+  {
+    "type": "overmap_terrain",
+    "id": "abandonedwarehouse",
+    "copy-from": "generic_city_building",
+    "name": "abandoned warehouse",
+    "sym": 119,
+    "color": "brown"
+  },
+  {
+    "type": "overmap_terrain",
+    "id": "abandonedwarehouse_roof",
+    "copy-from": "generic_city_building",
+    "name": "abandoned warehouse roof",
+    "sym": 119,
+    "color": "brown"
+}
+````
+## Adding to regional_map_settings
+Navigate to `data/json/regional_map_settings.json`
+
+This determines the spawn frequency and location of non-special buildings.
+Find the appropriate category for your building and add the om_terrain ID for the main floor only and a spawn weight.
```

Comment:
The entry should be the `overmap_special` or `city_building` ID, not the `overmap_terrain` ID for the main floor. If you provide an `overmap_terrain` ID, the game will take it and turn it into a `FakeSpecial_` internally, but it won't load your `city_building`/`overmap_special` definition, and thus won't have your associated roof.